back-end/hub-core/src/main/java/io/apicurio/hub/core/registry/IRegistry.java

```diff
@@ -0,0 +1,15 @@
+package io.apicurio.hub.core.registry;
+
+import io.apicurio.registry.rest.beans.ArtifactMetaData;
+import io.apicurio.registry.rest.beans.SearchedArtifact;
+
+import java.io.IOException;
+import java.util.Collection;
+
+public interface IRegistry {
```

Comment:
Some javadoc on this would be good.  I'm assuming this interface is how Studio interacts with an instance of Registry?  I wonder - do you anticipate multiple implementations of this?  Currently the only way for Studio to access Registry is via its REST API.

Reply:
I didn't know the REST API was the only way through, but coming from Spring I would have defined a no-op implementation for when the registry URL is not set. This could still be used for a MockRegistry implementation if necessary.

Reply:
Cool.  I don't mind the extra layer of abstraction.  It's probably not going to be needed, but it'll be nice to have if I'm wrong.